http/handler.go

```diff
@@ -177,6 +177,18 @@ func Handler(props *vault.HandlerProperties) http.Handler {
 			mux.Handle("/v1/sys/metrics", handleLogicalNoForward(core))
 		}
 
+		if props.ListenerConfig != nil && props.ListenerConfig.Profiling.UnauthenticatedPProfAccess {
+			for _, name := range []string{"goroutine", "threadcreate", "heap", "allocs", "block", "mutex"} {
```

Comment:
Some of these aren't supported by the existing code: threadcreate, allocs, block, and mutex.  I'll see if I can find out if they were excluded deliberately or not, and if not, add them to the existing sys handlers.

Reply:
I've added them pre-emptively.  @calvn do you know if they were omitted deliberately?

Reply:
The initial implementation was mostly a passthrough of the handlers exposed on the `net/http/pprof` package which includes cmdline, profile, symbol, and trace, but not the ones above since they were not exposed by it directly. 

I see that you're now routing them directly in here instead. Does it make `vault/logical_system_pprof.go` moot?

Reply:
`heap` and `goroutine` were already supported, I'm adding `allocs`, `block`, and `mutex`.  All of these are invoked in the same way, hence my question.

As to "does this obviate the need for logical_system_pprof" - no, because for authenticated requests we still want to go through our own request handlers for auth check, auditing, etc.

Reply:
Ah, I see that it still routes to those paths on the authenticated case. Would it make sense to add profiling on `[]string{"goroutine", "threadcreate", "heap", "allocs", "block", "mutex"}` for the authenticated scenario as well?

Reply:
Absolutely, think I meant to then forgot.  Thanks for suggestion.

Reply:
Added